### PR TITLE
encourage users to not copy their secrets on shell history

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
The `--password` arg may leak a secret into `ps`/`top` output or to shell history. We can encourage users to use a more secure way to supply their secrets. Although existing flags remain, we will mark them as hidden.

#### Release note

```release-note
mmctl will accept --password-file or --access-token-file for login secrets. Please consider updating your scripts accordingly
```

### CODE REVIEW

File: commands/auth.go [hunk 1]
```diff
@@ -127,10 +136,32 @@ func loginCmdF(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+	passwordFile, _ := cmd.Flags().GetString("password-file")
+	if password != "" && passwordFile != "" {
+		return errors.New("can not use two passwords at the same time")
```
Comment (streamer45) [inline commands/auth.go:141]:
```suggestion
		return errors.New("cannot use two passwords at the same time")
```

Same for the other occurrences.

File: commands/auth.go [hunk 2]
```diff
@@ -127,10 +136,32 @@ func loginCmdF(cmd *cobra.Command, args []string) error {
...
+	}
+	if passwordFile != "" {
+		b, err2 := ioutil.ReadFile(passwordFile)
+		if err2 != nil {
+			return err2
+		}
+		password = strings.TrimSpace(string(b))
```
Comment (streamer45) [inline commands/auth.go:148]: Wondering if this should become something like a `readStringFromFile` util function.

Comment (isacikgoz) [reply commands/auth.go:148]: Makes sense 👍

**Review by streamer45 — APPROVED** [review]
Nice :+1:

**Pull request merged**